mailpoet/lib/AdminPages/Pages/AutomationEditor.php

```diff
@@ -108,4 +109,12 @@ private function buildRegistry(): array {
     }
     return ['steps' => $steps];
   }
+
+  private function buildContext(): array {
+    $data = [];
+    foreach ($this->registry->getContextFactories() as $key => $factory) {
+      $data[$key] = $factory();
+    }
```

Comment:
That's a good point! I'm just thinking about how to address a failure...

Should we display a notice in the editor that loading some integration failed? I guess that might be the best, but it will require propagating the errors to the UI somehow.

To add more context to the usage of these contexts, I think we could/should make it async so that the editor loads as fast as possible and the contexts are fetched with an API request asynchronously. Then, I guess, that async request could also contain an array of integration errors that we could convert into notices in the UI.

If that sounds good to you, I will compile it into a ticket so it doesn't get lost here. What do you think?

Reply:
That sounds like a good solution to me, especially ensuring the UI loads as fast as possible. 

Initially I was thinking that maybe some simple logging would be sufficient in the case of errors, but realistically we might have people writing integrations who may not think to check logs. It might be good to include some logs with more details (e.g. a stack trace) that would be messy to have in the UI.

tldr that all sounds great to me!